### DISCUSSION TITLE
Using 3P APIs to fetch GST details rather than asking user to enable

### DISCUSSION THREAD

**Discussion by SanjanaIL** [ONDC-Official/ONDC-FIS-Specifications#38] [General]
![image](https://github.com/ONDC-Official/ONDC-FIS-Specifications/assets/142986416/2dab9f2b-1fb9-4d6c-b1dc-47d4c0538737)

1) Is there a way/API to take consent from the user and enable this access on the GSTN portal? Or can we get the GST details/invoices from a 3rd Party API from vendors like Karza and share it? 
2) Can we know the details to be shared with the Lenders so that we can use some third party APIs to fetch those details?

**Comment by 92shreyansh** [comment]
@Antriksh-Parmar

**Reply by Antriksh-Parmar** [reply]
point 1. Unfortunately there's no way to enable it via APIs. Will you be able to retrieve GSTR1, 2A and 3B via their APIs? 
point 2. The details to be shared with the lenders are captured in the BRD. I'd recommend you to refer to the MSME use case.

**Reply by pawank** [reply]
In the https://ondc-official.github.io/ONDC-FIS-Specifications/#Get%20the%20Invoice%20based%20Org%20form form API, the form is expecting GSTR1, GSTR2A , GSTR3B files but there is no mention of the following 

1. The file type of these files - JSON or Excel
2. Are all of these mandatory or GSTR1 is sufficient?
3. The time period for which the data is to be downloaded/fetched from GST portal?
